types/office-js-preview/index.d.ts

```diff
@@ -476,6 +516,114 @@ declare namespace Office {
         */
         value: T;
     }
+    /**
+     * Message used in the visibility mode changed event.
+     */
+    interface VisibilityModeChangedMessage {
+        /**
+         * Visibility changed state.
+         */
+        visibilityMode: Office.VisibilityMode;
+    }
+    /**
+     * Function type to turn off the event.
+     */
+    type RemoveEventListener = () => Promise<void>;
+    /**
+     * Represents add-in level functionality for operating or configuring various aspects of the add-in.
+     */
+    interface Addin {
+        /**
+         * Set the startup behavior for the add-in for when the document is opened next time.
+         * @param behavior Specifies startup behavior of the add-in.
+         */
+        setStartupBehavior(behavior: Office.StartupBehavior): Promise<void>;
+        /**
+         * Get the current startup behavior for the add-in.
+         */
+        getStartupBehavior(): Promise<Office.StartupBehavior>;
+        /**
+         * Shows the task pane associated with the add-in.
+         * @returns A promise that is resolved when the UI is shown.
+         */
+        showAsTaskpane(): Promise<void>;
+        /**
+         * Hides the task pane.
+         * @returns A promise that is resolved when the UI is hidden.
+         */
+        hide(): Promise<void>;
+        /**
+         * Adds a listener for the VisbilityModeChanged event.
+         * @param listener The listener function that is called when the event is emitted. This function takes in a message for the receiving component.
+         * @returns A promise that resolves when the listener is added.
+         */
+        onVisibilityModeChanged(
+            listener: (message: VisibilityModeChangedMessage) => void,
+        ): Promise<RemoveEventListener>;
+    }
+    /**
+     * Ribbon interface contains all the functionality provided to manage the state of the OFfice ribbon.
```

Comment:
Grammar nit: It sounds a little awkward to say "Ribbon interface contains". How about "An interface that contains"?
That way, we can avoid having to worry about denoting the interface name with code ticks too.